framework/src/meshgenerators/BlockDeletionGenerator.C

```diff
@@ -22,18 +22,27 @@ BlockDeletionGenerator::validParams()
 
   params.addClassDescription(
       "Mesh generator which removes elements with the specified subdomain ID");
-  params.addRequiredParam<SubdomainID>("block_id", "The block to be deleted");
+  params.addParam<std::vector<SubdomainID>>("block_ids", "The list of blocks to be deleted");
+  params.addDeprecatedParam<SubdomainID>(
+      "block_id", 0, "The block to be deleted. Use block_ids instead");
```

Comment:
While you're at it, you should really just use

`params.addParam<std::vector<SubdomainName>>("block")`, to be consistent with all of our current block restrictable objects.